### DISCUSSION TITLE
Is it possible to create a scratchpad file manager window on startup without messing up my regular window rules?

### DISCUSSION THREAD

**Discussion by SpinningVinyl** [i3/i3#5655] [Q&A] [answered]
Hi everyone, I am currently test-driving i3wm on my old laptop and mostly loving it so far. However, I have a problem that I can't solve on my own.

I have to use a pretty annoying app that only accepts reports via drag-and-drop. I have to upload files using this app several times a day. I have this app assigned to workspace 1, while my file manager is assigned to workspace 4.

So far, every time I have to upload a report, it goes like this:

1. Launch a new instance of my GUI file manager (I use Nemo).
2. The file manager opens on workspace 4, so move to WS4 and send the file manager window to WS1.
3. Move back to WS1, select the file manager window and toggle floating.
4. Drag report files into the annoying app.
5. Close the file manager window.

The whole process is kinda annoying, and I would like i3 to automatically spawn a file manager window, assign a mark to it  and send it to the scratchpad whenever I log into my computer. However, I don't know how to do it without messing up my regular window rules.

So far I have tried the following:

```
exec --no-startup-id nemo; for_window [urgent=latest] mark 'fs', move scratchpad 
```

I used [urgent=latest] because Nemo actually tries to steal focus when it launches. It does not work though, probably because Nemo takes its sweet time to load and its window is not ready by the time i3 executes the next command.

Is there any way to accomplish this? The best idea I came up with is to launch Nemo via a bash script that would grab its PID and then use i3-msg to match the window by the PID (since it properly sets _NET_WM_PID), but as far as I understand i3wm does not allow matching windows by the PID.

**Comment by orestisfl** [comment, marked as answer]
> I used [urgent=latest] because Nemo actually tries to steal focus when it launches. It does not work though, probably because Nemo takes its sweet time to load and its window is not ready by the time i3 executes the next command.

It's an i3 bug, `for_window` rules don't work with `urgent`: https://github.com/i3/i3/issues/5658

Interesting how two people are affected by this in the span of an hour: https://github.com/i3/i3/discussions/5656

**Comment by SpinningVinyl** [comment]
Thank you Orestis. That is a strange coincidence indeed. I have been trying to make it work for days, so it's nice to know that a fix is on the way :)
